src/parachain/vaults.ts

```diff
@@ -300,24 +328,39 @@ export class DefaultVaultsAPI extends DefaultTransactionAPI implements VaultsAPI
         return Promise.all(
             vaultsMap
                 .filter((v) => v[1].isSome)
-                .map((v) => this.parseVault(v[1].value as DefaultVault, this.btcNetwork))
+                .map((v) => this.parseVault(v[1].value as VaultRegistryVault, this.btcNetwork))
         );
     }
 
-    async get(vaultId: AccountId): Promise<VaultExt<BitcoinUnit>> {
-        const head = await this.api.rpc.chain.getFinalizedHead();
-        const vault = await this.api.query.vaultRegistry.vaults.at(head, vaultId);
-        if (!vault.isSome) {
-            return Promise.reject(`No vault registered with id ${vaultId}`);
+    async get(
+        vaultAccountId: AccountId,
+        collateralCurrencyIdLiteral: CollateralIdLiteral
+    ): Promise<VaultExt<BitcoinUnit>> {
+        try {
+            const head = await this.api.rpc.chain.getFinalizedHead();
+            const collateralCurrency = currencyIdToMonetaryCurrency(
+                newCurrencyId(this.api, collateralCurrencyIdLiteral)
+            ) as CollateralCurrency;
+            const vaultId = newVaultId(this.api, vaultAccountId.toString(), collateralCurrency, this.wrappedCurrency);
+            const api = await this.api.at(head);
```

Comment:
Not sure I fully understand this, don't you have an `api` instance already?

Reply:
Queries such as `api.query.vaultRegistry.vaults.at(...)` have become deprecated. We first need to get an `api` object for a specific block and then query. I tried it here to see if it works, but I can keep it uniform with the rest until I change everywhere